code/HabitTracker/app/src/main/java/com/example/habittracker/DatabaseManager.java

```diff
@@ -363,6 +374,11 @@ public void onComplete(@NonNull Task<DocumentSnapshot> task) {
         });
     }
 
+    /**
+     * Check if username exists
+     * @param username
+     * @param callback
+     */
     public void userExists(String username, BooleanCallback callback){
```

Comment:
public interface UserExistsCallback {
  void onCallbackSuccess(String username)
  void onCallbackFailed()
}

Please use this UserExistsCallback interface instead of BooleanCallbackInterface.

This is because I need the username returned to me.

Reply:
oh sorry, I will commit the new change for you